app/src/org/commcare/views/widgets/GregorianDateWidget.java

```diff
@@ -46,21 +46,20 @@
     private static final int EMPTY_MONTH_ENTRY_INDEX = 12;
 
     private List<String> monthList;
-    private final int maxYear;
     private final long todaysDateInMillis;
     private long timeBeforeCalendarOpened;
 
     private final CalendarFragment myCalendarFragment;
     private final FragmentManager fm;
 
-    public static final int MINYEAR = 1900;
+    public static final int MIN_YEAR = 1900;
+    public static final int MAX_YEAR = 2100;

```

Comment:
Fix number 1: Instead of setting the max year based upon an offset from the currently-selected year, just use the same hard-coded max that the default Android date-picker uses